Src/SwqlStudio/Utils/CommandLineGenerator.cs

```diff
@@ -7,25 +9,45 @@ namespace SwqlStudio.Utils
 {
     internal static class CommandLineGenerator
     {
-        public static string GetQueryForCurlCmd(string query, ConnectionInfo connection)
+        public static string GetQueryForCurlCmd(string query, ConnectionInfo connection, PropertyBag parameters)
         {
             string creds = QuoteForCmd($"{connection.UserName}:{connection.Password}");
             return $"curl.exe -k -u {creds} {GetUrlForQuery(query, connection)}";
         }
 
-        public static string GetQueryForCurlBash(string query, ConnectionInfo connection)
+        public static string GetQueryForCurlBash(string query, ConnectionInfo connection, PropertyBag parameters)
         {
             string creds = $"{connection.UserName}:{connection.Password}";
             var url = GetUrlForQuery(query, connection);
             Func<string, string> q = QuoteForBash;
             return $"curl -k -u {QuoteForBash(creds)} {QuoteForBash(url)}";
         }
 
-        public static string GetQueryForPowerShellGetSwisData(string query, ConnectionInfo connection)
+        public static string GetQueryForPowerShellGetSwisData(string query, ConnectionInfo connection, PropertyBag parameters)
         {
             Func<string, string> q = QuoteForPowerShell;
+            var paramsObjectString = GetParamsObjectInPowershellFormat(parameters);
             return $"Get-SwisData (Connect-Swis -Hostname {connection.Server} -Username {q(connection.UserName)} " +
-                   $"-Password {q(connection.Password)}) -Query {q(CollapseWhitespace(query))}";
+                   $"-Password {q(connection.Password)}) -Query {q(CollapseWhitespace(query))} -Parameters {paramsObjectString}";
+        }
+
+        private static string GetParamsObjectInPowershellFormat(PropertyBag parameters)
+        {
+            var result = new StringBuilder();
+            var parametersCount = parameters.Count;
+            var currentParameterIndex = 0;
+            foreach (var param in parameters)
+            {
+                decimal tempNum = 0;
+                currentParameterIndex++;
+                var isNumber = decimal.TryParse(param.Value.ToString(), out tempNum);
+                var format = "{0}='{1}'";
+                var isLastElement = currentParameterIndex == parametersCount;
+                format = isLastElement ? format : string.Concat(format, ";");
+                result.Append(string.Format(format, param.Key, param.Value));
```

Comment:
Rather than mess around with `StringBuilder` and `.Append` and conditionally inserting `;`, I would just format each parameter separately and combine them all with `string.Join`. There are never going to be enough parameters for the performance difference to matter.